helm/nginx-ingress-controller-app/templates/controller-hpa.yaml

```diff
@@ -1,7 +1,5 @@
-{{- if and .Values.controller.autoscaling.enabled (or (eq .Values.controller.kind "Deployment") (eq .Values.controller.kind "Both")) -}}
-{{- if not .Values.controller.keda.enabled }}
-
-apiVersion: {{ .Values.controller.autoscaling.apiVersion }}
+{{- if and .Values.controller.autoscaling.enabled (or (eq .Values.controller.kind "Deployment") (eq .Values.controller.kind "Both")) (not .Values.controller.keda.enabled) -}}
+apiVersion: {{ ternary "autoscaling/v2" "autoscaling/v2beta2" (semverCompare ">=1.23.0-0" .Capabilities.KubeVersion.Version) }}
```

Comment:
We could still check for `.Values.controller.autoscaling.apiVersion` first. And instead of checking `.Capabilities.KubeVersion.Version` maybe use `.Capabilities.APIVersions.Has "autoscaling/v2"`. See for [example in upstream](https://github.com/kubernetes/ingress-nginx/blob/d7674e43230274a485ff90054383083ea4280ef8/charts/ingress-nginx/templates/admission-webhooks/job-patch/job-patchWebhook.yaml#L20).

Reply:
I don't think this makes sense because the rest of the template is already made up for `autoscaling/v2beta2` or higher and not working with `autoscaling/v2beta1` or even `autoscaling/v1`. So even if you could override it, it would not work due to the `.spec`.

Reply:
Regarding `Capabilities.APIVersions.Has`: Yeah, I know. But I just sticked to most of the other manifests, they are still using the hard coded comparison.

Reply:
Ok, I was thinking about the alignment with upstream. There the setting for `.Values.controller.autoscaling.apiVersion` is documented and customers might be confused when we differ.

Reply:
Regarding Capabilities.APIVersions.Has: You want to know the `apiVersion`, so you should just ask for the `apiVersion` :) Since upstream also uses this, they should be fine when we request a PR. But I am not to opinionated about this..

Reply:
I will also drop the support for this field in my PR to upstream, I guess. As said, I don't think there is a real value addition for now. Also never saw this way of determining the API version before. 🤷

Reply:
Totally agree, should be dropped from upstream, so ok all good here for me. Regarding `apiVersion` check: Kubernetes installations with same version might still differ in provided api-resources and versions because of the [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/). I guess that's why Helm provides this check and I think it should be best practice. But I don't want to block if you hesitate.

Reply:
Ahh! Yeah, you're right! Gonna use the `Has`. Thank you so much! :)

Reply:
Thank you very much for working on this PR! :)